Deferred/Deferred.swift

```diff
@@ -116,16 +159,43 @@ public struct Deferred<Value> {
 }
 
 extension Deferred {
+    /**
+    Checks for and returns a determined value.
+    
+    :returns: The determined value, if already filled, or `nil`.
+    */
     public func peek() -> Value? {
         return wait(.Now)
     }
-
+    
+    /**
+    Waits for the value to become determined, then returns it.
+    
+    This is equivalent to unwrapping the value of calling `wait(.Forever)`, but
+    may be more efficient.
+    
+    This getter will unnecessarily block execution. It might be useful for
+    testing, but otherwise it should be strictly avoided.
+    
+    :returns: The determined value.
+    */
     public var value: Value {
         return unsafeUnwrap(wait(.Forever))
     }
 }
 
 extension Deferred {
+    /**
+    Begins another asynchronous operation with the deferred value once it
+    becomes determined.
+    
+    :param: queue A dispatch queue for starting the new operation on.
+    :param: transform A function to start a new deferred given the recieving
+    value.
+    
+    :returns: A new deferred value that is determined once the recieving
```

Comment:
typo "recieving"

More generally, I found this `:returns:` sentence really hard to follow. I think it might help to not talk about resolving the returned deferred. Maybe just, "The deferred value returned by `transform`."?